src/devices/machine/i8291a.cpp

```diff
@@ -0,0 +1,1082 @@
+// license:BSD-3-Clause
+// copyright-holders: Sven Schnelle
+/*********************************************************************
+
+ i8291a.cpp
+
+ Intel 8291A GPIB controller
+
+ *********************************************************************/
+
+#include "emu.h"
+
+
+#include "i8291a.h"
+
+#include "logmacro.h"
+#define LOG_STATE_MASK  (LOG_GENERAL << 1)
+#define LOG_STATE(...)  LOGMASKED(LOG_STATE_MASK, __VA_ARGS__)
+#define LOG_REG_MASK    (LOG_STATE_MASK << 1)
+#define LOG_REG(...)    LOGMASKED(LOG_REG_MASK, __VA_ARGS__)
+#define LOG_INT_MASK    (LOG_REG_MASK << 1)
+#define LOG_INT(...)    LOGMASKED(LOG_INT_MASK, __VA_ARGS__)
+#define LOG_CMD_MASK	(LOG_INT_MASK << 1)
+#define LOG_CMD(...)	LOGMASKED(LOG_CMD_MASK, __VA_ARGS__)
+#undef VERBOSE
+#define VERBOSE (LOG_GENERAL)//|LOG_REG_MASK|LOG_CMD_MASK|LOG_STATE_MASK|LOG_STATE_MASK)
+
+DEFINE_DEVICE_TYPE(I8291A, i8291a_device, "i8291a", "I8291A")
+
+i8291a_device::i8291a_device(const machine_config &mconfig, const char *tag,
+		device_t *owner, uint32_t clock) :
+		device_t(mconfig, I8291A, tag, owner, clock),
+		m_int_write_func(*this),
+		m_dreq_write_func(*this),
+		m_trig_write_func(*this),
+		m_eoi_write_func{*this},
+		m_dav_write_func{*this},
+		m_nrfd_write_func{*this},
+		m_ndac_write_func{*this},
+		m_srq_write_func{*this},
+		m_dio_write_func{*this},
+		m_dio_read_func{*this}
```

Comment:
Why are you using uniform initialiser syntax for some class type members but not for others?  It's jarring.